telmos_goods.py

```diff
@@ -179,80 +156,32 @@ def telmos_goods(delta_root, tmfs_root, tel_year, tel_id, tel_scenario,
                                         nrtf_array.PERIOD==
                                                 int(base_year)+2000][nrtf_col].values[0]))
             # For up to hgv_count use the following
-            new_forecast_array[f_key][:hgv_count,:hgv_count] = (
-                    forecast_goods_array[f_key][:hgv_count,:hgv_count] * 
+            new_forecast_array[f_key][:zone_count,:zone_count] = (
+                    forecast_goods_array[f_key][:zone_count,:zone_count] * 
                     ((goods_total["%s_base" % f_key] * goods_totals["TEL_%s" % goods_type]) / 
                      (goods_total["%s_forecast" % f_key] * goods_totals["BASE_%s" % goods_type]))
                     )
-        # Print these files as stack
-        if do_output:
-            df = pd.DataFrame(new_forecast_array[f_key]).stack().reset_index()
-            df.loc[:,:"level_1"] = df.loc[:,:"level_1"] + 1
-            df.to_csv(os.path.join(tel_filebase, filename), index=False, 
-                      header=False, float_format="%.3f")
-            log_func("Goods file saved to %s" % str(os.path.join(tel_filebase, filename)))
+        # Print these files stacked
+        df = pd.DataFrame(new_forecast_array[f_key]).stack().reset_index()
+        df.loc[:,:"level_1"] = df.loc[:,:"level_1"] + 1

```

Comment:
This is a bit gross. Might help to make it more explicit what's happening in the `DataFrame`:
```suggestion
        # Get matrix in "long" form to write as CSV 
        df = pd.DataFrame(new_forecast_array[f_key]).stack().reset_index()
        df.columns = ['I', 'J', 'V']
        
        # Change zone numbers from 0-based to 1-based
        df['I'] += 1
        df['J'] += 1
```